### PR TITLE
[webview_flutter] Updates example app and convert integration tests for new implementation

### PR DESCRIPTION
**Pull request by bparrishMines**
* Updates `main.dart` to use new implementation.
* Moves legacy integration test to `legacy` folder.
* Updates old integration test with new implementation.
* Update `example/test/main_test.dart`. 
* Adds files to import `webview_flutter_android` and `webview_flutter_wkwebview` classes.

Part of https://github.com/flutter/flutter/issues/94051

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [relevant style guides] and ran [the auto-formatter]. (Unlike the flutter/flutter repo, the flutter/plugins repo does use `dart format`.)
- [x] I signed the [CLA].
- [x] The title of the PR starts with the name of the plugin surrounded by square brackets, e.g. `[shared_preferences]`
- [x] I listed at least one issue that this PR fixes in the description above.
- [ ] I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy], or this PR is [exempt from version changes].
- [ ] I updated `CHANGELOG.md` to add a description of the change, [following repository CHANGELOG style].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/plugins/blob/main/CONTRIBUTING.md
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[relevant style guides]: https://github.com/flutter/plugins/blob/main/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[exempt from version changes]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#version-and-changelog-updates
[following repository CHANGELOG style]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#changelog-style
[the auto-formatter]: https://github.com/flutter/plugins/blob/main/script/tool/README.md#format-code
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests

### CODE REVIEW

File: packages/webview_flutter/webview_flutter/example/integration_test/webview_flutter_test.dart [hunk 1]
```diff
@@ -227,522 +156,30 @@ Future<void> main() async {
...
-          javascriptMode: JavascriptMode.unrestricted,
-        ),
-      ),
-    );
-
-    final String customUserAgent2 = await _getUserAgent(controller);
-    expect(customUserAgent2, defaultPlatformUserAgent);
-  });
-
-  group('Video playback policy', () {
```
Comment (stuartmorgan) [inline packages/webview_flutter/webview_flutter/example/integration_test/webview_flutter_test.dart:324]: Why are we losing this whole test group?

File: packages/webview_flutter/webview_flutter/example/integration_test/webview_flutter_test.dart [hunk 2]
```diff
@@ -1210,86 +462,6 @@ Future<void> main() async {
       await expectLater(controller.currentUrl(), completion(primaryUrl));
     },
   );
-
-  testWidgets(
-    'clearCache should clear local storage',
```
Comment (stuartmorgan) [inline packages/webview_flutter/webview_flutter/example/integration_test/webview_flutter_test.dart:1215]: Why are we losing this test?

**Review by stuartmorgan — APPROVED** [review]
LGTM! I left a couple questions about dropped tests; I'm assuming there's a good reason (e.g., we don't think those are valuable enough to duplicate in the app-facing integration tests?), but I thought we should leave an explicit record of why.

File: packages/webview_flutter/webview_flutter/example/integration_test/webview_flutter_test.dart (hunk 2 above)
Comment (bparrishMines) [reply packages/webview_flutter/webview_flutter/example/integration_test/webview_flutter_test.dart:1215]: `clearCache` and `clearLocalStorage` are now two separate methods now. I'll add this test back and make it a test for just calling `clearLocalStorage` instead.

File: packages/webview_flutter/webview_flutter/example/integration_test/webview_flutter_test.dart (hunk 1 above)
Comment (bparrishMines) [reply packages/webview_flutter/webview_flutter/example/integration_test/webview_flutter_test.dart:324]: The methods to set these are platform specific and I didn't want to add any platform specific testing in the app-facing package.

Comment (bparrishMines) [reply packages/webview_flutter/webview_flutter/example/integration_test/webview_flutter_test.dart:324]: I could add them back though. And then we would have testing for importing and accessing platform specific features.

**Pull request merged**